src/commands/mod.rs

```diff
@@ -2592,29 +2609,37 @@ don't make any changes, then the operation will be aborted.
         .store()
         .get_tree(&RepoPath::root(), &tree_id)?;
 
-    let first_template = description_template_for_cmd_split(
-        tx.base_workspace_helper(),
-        "Enter commit description for the first part (parent).",
-        commit.description(),
-        &base_tree,
-        &middle_tree,
-    )?;
-    let first_description = edit_description(tx.base_repo(), &first_template, command.settings())?;
+    let (first_description, second_description) = if args.empty_child {
+        (commit.description().to_owned(), String::new())
+    } else if args.empty_parent {
+        (String::new(), commit.description().to_owned())
+    } else {
+        let first_template = description_template_for_cmd_split(
+            tx.base_workspace_helper(),
+            "Enter commit description for the first part (parent).",
+            commit.description(),
+            &base_tree,
+            &middle_tree,
+        )?;
+        let first_description =
+            edit_description(tx.base_repo(), &first_template, command.settings())?;
+        let second_template = description_template_for_cmd_split(
+            tx.base_workspace_helper(),
+            "Enter commit description for the second part (child).",
+            commit.description(),
+            &middle_tree,
+            &commit.tree(),
+        )?;
+        let second_description =
+            edit_description(tx.base_repo(), &second_template, command.settings())?;
+        (first_description, second_description)
+    };
     let first_commit = tx
         .mut_repo()
         .rewrite_commit(command.settings(), &commit)
         .set_tree(tree_id)
         .set_description(first_description)
```

Comment:
In the case of `--empty-parent` (only), this commit should get a new change_id while `second_commit` below shouldn't.

Reply:
I realized this also raises the question of whether branches should go with the child even on `--empty-parent`. It's a bit arguable, but I think for `jj split` the answer should be "yes". So, just changing where the id goes and not worrying about the branches would do the right thing.

Reply:
> In the case of `--empty-parent` (only), this commit should get a new change_id while `second_commit` below shouldn't.

This is done in the second commit of this issue.

Reply:
Thank you!